docs/melange-book/counter.md

```diff
@@ -0,0 +1,176 @@
+# Counter
+
+We're going build the classic frontend starter app, the counter, using
+[ReasonReact](https://reasonml.github.io/reason-react/). Let's clone the
+tutorial template and initialize it:
+
+    git clone https://github.com/melange-re/melange-tutorial
+    cd melange-tutorial
+    make init
+
+To start webpack, run `make serve`. The app will be served at
+http://localhost:8080/. In another terminal window, start the Melange compiler
+in watch mode by running `make watch`.
+
+Open `Index.re` and you'll see this:
+
+```reason
+module App = {
+  [@react.component]
+  let make = () => <div> {React.string("welcome to my app")} </div>;
+};
+```
+
+This is just about the simplest component you can make. Unlike in normal React,
+we must wrap strings inside function calls to convert them to the
+`React.element` type. This is exactly what the `React.string` function does--if
+you hover over it, you'll see that it displays the type `string =>
+React.element`. The `make` function itself has the type `unit => React.element`,
+meaning it takes `()` as the only argument and returns an object of type
+`React.element`.
+
+What's with the `module` business? OCaml's
+[modules](https://cs3110.github.io/textbook/chapters/modules/modules.html) are
+somewhat like JavaScript modules, with one (of many) notable differences being
+that there can be multiples modules inside a single file. For now, you just need
+to know that all components in ReasonReact are modules.
+
+A little bit further down, we make use of the `App` component:
+
+```reason
+let node = ReactDOM.querySelector("#root");
+switch (node) {
+| Some(root) => ReactDOM.render(<App />, root)
+| None => Js.Console.error("Failed to start React: couldn't find the #root element")
+};
+```
+
+`React.querySelector("#root")` returns an `option(Dom.element)`, meaning that if
+it doesn't find the element, it returns `None`, and if it does find the element,
+it returns `Some(Dom.element)`, i.e. the element wrapped in the `Some`
+constructor. The `switch` expression (despite the name, don't confuse it for the
+JavaScript construct of the same name), and allows you to succinctly express:
+
+- If `node` is `Some(Dom.element)`, render the `App` component to the DOM
+- Otherwise if `node` is `None`, log an error message
+
+Don't worry if this doesn't quite make sense yet, we'll talk more about `option`
```

Comment:
Does the first part add any value?

```suggestion
We'll talk more about `option`
```

Reply:
Removed in https://github.com/melange-re/melange-re.github.io/pull/93/commits/2fbcdbeac74c78b6b871e42bab09cb7ebec86218